Makefile

```diff
@@ -19,6 +19,9 @@ proto:
 test: testdeps
 	go test -cpu 1,4 -timeout 7m google.golang.org/grpc/...
 
+testsubmodule: testdeps
+	cd security/advancedtls && go test -cpu 1,4 -timeout 7m google.golang.org/grpc/security/advancedtls/... && cd ../..
```

Comment:
(I'm not very sure about this)

I think the `cd ../..` is not necessary at the end, because make runs the commands in a sub-shell (??).

But keeping it would also be OK.

Reply:
Yes you are right. Make runs every line in a sub shell. I added that just to keep every sub shell clean after running. Let's keep it if you don't mind :-)

Reply:
> I added that just to keep every sub shell clean after running

If `go test` fails, `cd ../..` will not execute (because of `&&`), and you will be "left in `security/advancedtls`".  But of course, because it's a sub-shell, it doesn't matter.  I'd just remove it for simplicity.

Really, `go` should accept a parameter to specify the module root, to avoid the need to `cd` at all, but no such flag exists.

Reply:
Thanks for pointing that out! I have updated the code accordingly.

Reply:
Thanks for pointing that out! I have updated the code accordingly.